updated grammar with text

byronbytes opened the pull request:
It's still being improved, but some of it sounds gramatically incorrect.

byronbytes commented:
Do not merge this PR yet, I'm still working on it

byronbytes commented:
Okay I just made the final changes now it can be merged
if you have any suggestions let me know

ryo-ma commented:
I'm not a native English speaker.
If you like, please tell me the reason for each change

byronbytes commented:
> Okay I just made the final changes now it can be merged if you have any suggestions let me know

That's okay! I can help

I changed the tense (Past, present, future) of the achievements since for example "High Star" doesn't make sense as an achievement, but "Many Stars" makes more sense.

I also changed Deep to Omega, since deep does not describe an achievement and doesn't make much sense.

I also changed 'god' to 'godly'

and I updated the Star category with new achievement names.

spenserblack commented:
This will affect many users, and some users might not anticipate this change. I think it might be best to wait until other users have offered their opinion on this PR.
(I'll try my best to also put my comments in Japanese for the sake of the repo owner)

影響する利用者が多いはずです。この変化を期待していない利用者がいるでしょう。利用者の意見をもらうまで待つ方がいいでしょうか？

ryo-ma commented:
@spenserblack What do you think of each change?

spenserblack commented:
It's mostly good to me, but I'm curious about the word "Omega" for overseas cultures. What do they think of "Omega"? "Omega" is somewhat common in America.
I'm not sure why "Super Star" was changed into "Little Stargazer". Why not "Super Stargazer"?
I think that the meanings of "Numerous", "Many", and "Multiple" aren't too different. Words that are more different might be better. Maybe words like "galactic"?


これは私の個人的な意見です。大部分がいいと思います。でも、外国人にとって「OMEGA」とはどうでしょうか。明らかな意味ですか？アメリカには「OMEGA」は少し通用です。
さらに、「SUPER STAR」は「LITTLE STARGAZER」に代えるのはなぜですか。「SUPER STARGAZER」のほうが良くないですか？
「NUMEROUS」と「MANY」と「MULTIPLE」は、意味があまり違いません。もっと違う意味の言葉は、そのほうがいいかもしれません。「GALACTIC」などとはどうでしょう。

byronbytes commented:
> It's mostly good to me, but I'm curious about the word "Omega" for overseas cultures. What do they think of "Omega"? "Omega" is somewhat common in America. I'm not sure why "Super Star" was changed into "Little Stargazer". Why not "Super Stargazer"? I think that the meanings of "Numerous", "Many", and "Multiple" aren't too different. Words that are more different might be better. Maybe words like "galactic"?
> 
> これは私の個人的な意見です。大部分がいいと思います。でも、外国人にとって「OMEGA」とはどうでしょうか。明らかな意味ですか？アメリカには「OMEGA」は少し通用です。 さらに、「SUPER STAR」は「LITTLE STARGAZER」に代えるのはなぜですか。「SUPER STARGAZER」のほうが良くないですか？ 「NUMEROUS」と「MANY」と「MULTIPLE」は、意味があまり違いません。もっと違う意味の言葉は、そのほうがいいかもしれません。「GALACTIC」などはどうでしょう。

Hmm, it was a little bit difficult to come up with words lol 😆, but I do like galactic, like it's out of this world!

spenserblack commented:
#136 Gave me an idea. Just like themes, there could be a `?text=` argument. This allows for different languages to be chosen, but also allows the user to choose the original text instead of the updated text (`?text=original`). This is less of a breaking change for users IMO.

#136 から考えが出来ました。テーマの同様、`?text=`で言葉を選べ方が出来ます。それで利用者は別の言語とか元の言葉とか選べられます。

byronbytes commented:
> #136 Gave me an idea. Just like themes, there could be a `?text=` argument. This allows for different languages to be chosen, but also allows the user to choose the original text instead of the updated text (`?text=original`). This is less of a breaking change for users IMO.
> 
> #136 から考えが出来ました。テーマの同様、`?text=`で言葉を選べ方が出来ます。それで利用者は別の言語とか元の言葉とか選べられます。

Even better, with the addition of that feature you could add localization support for other languages (Japanese)

3kh0 reviewed (approved):
Translations look good!

byronbytes commented:
I was thinking, instead of using words for the ranking, how about using things such as Bronze, Silver, Gold, Platnium, etc. I was looking back on my edits and some of them still feel "off" in a way, as in they were still gramatically incorrect in the given context.

(also thank you for replying, i almost forgot about this)

byronbytes commented:
> Standardizing to Bronze, Silver, etc. would certainly make translations easier!
> 
> ```js
> const en = { silver: 'silver' }
> const jp = { silver: '銀' }
> const locales = { en, jp }
> const text = locales[locale][level]
> ```
> 
> However, I think that unique text for each trophy is more fun, so there's a trade-off.
> 
> Part of the problem is that some adjectives are synonyms -- there isn't a _strong_ difference between "many stars" and "multiple stars". Perhaps instead of adjectives, it should be nouns that change. It's much more difficult to think up appropriate wording, but I think can result in a much clearer distinction and more "natural-sounding" English. For example, the levels for number of followers could be
> 
> 1. influencer
> 2. celebrity
> 3. deity

I believe this is a much better idea, since trying to change the adjective is a lot more difficult, due to trying to find the right word for both the noun / adjective to match. If worse goes to worst, there could just be a way for people to create custom text formats, which would go hand in hand with this suggestion

> #136 Gave me an idea. Just like themes, there could be a `?text=` argument.

People could create PR's with their versions (more preferrably themed as in it has a consistant naming scheme, so it's not random) and if it's good then they can push it.

byronbytes commented:
How did this get closed?

3kh0 commented on src/trophy.ts:
```diff
@@ -149,21 +149,23 @@ export class Joined2020Trophy extends Trophy{
     this.filterTitles = ["Joined2020"];
     this.bottomMessage = "Joined 2020"
     this.hidden = true;
+    // question, why are you only awarded for making an account in 2020? i made an account in 2017~ so shouldn't it be based on how many years?
+    // example: milestone every 2-3 years 
```
Yeah what is up with that anyway

AryanP45 replied:
> I don't know for sure, but this repository was created in 2020, so maybe that's related?

I think so too

ice1000 commented on src/trophy.ts:
```diff
@@ -137,7 +137,7 @@ export class AllSuperRankTrophy extends Trophy{
 export class Joined2020Trophy extends Trophy{
   constructor(score: number){
-    const rankConditions = [
+    const rankConditions = [ 
```
This new space does not seem right

spenserblack replied:
Good catch!
@byronbytes For the text editor you use, I'd recommend installing the [EditorConfig](https://editorconfig.org/) plugin, so that [the `.editorconfig` file](https://github.com/ryo-ma/github-profile-trophy/blob/7b8a6a4100893bbf1c721e563c6d07e04c79e479/.editorconfig) will prevent formatting issues like this.

byronbytes commented:
Happy 2023 everyone!
Decided to start on a complete previous code rewrite on all my features, added a lot of notes for better clarity, and overall improvements. 

For some reason the PR closed again.

spenserblack commented:
>For some reason the PR closed again

FYI PRs are auto-closed whenever the base branch or fork is deleted :slightly_smiling_face:

byronbytes commented:
> > For some reason the PR closed again
> 
> FYI PRs are auto-closed whenever the base branch or fork is deleted 🙂

Good to know, I keep trying to push changes from another branch to the `patch-2` one and I'm guessing that's why it keeps messing up.

asmpro7 commented on src/trophy.ts:
```diff
@@ -108,13 +108,13 @@ export class MultipleLangTrophy extends Trophy{
     ];
     super(score, rankConditions);
     this.title = "MultiLanguage";
-    this.filterTitles = ["MultipleLang", "MultiLanguage"];
```
Good one this

asmpro7 commented on src/trophy.ts:
```diff
@@ -130,14 +130,37 @@ export class AllSuperRankTrophy extends Trophy{
-    this.filterTitles = ["AllSuperRank"];
-    this.bottomMessage = "All S Rank"
+    this.filterTitles = ["AllSuperRank", "AllSRank"];
+    this.bottomMessage = "All S Rank Trophies"
```
I think we should make them shorter

asmpro7 reviewed (commented):
My Opinion about some changes

ice1000 commented on src/trophy.ts:
```diff
@@ -40,7 +40,7 @@ export class Trophy {
     }
   }
-  private calculateNextRankPercentage() {
+  private calculateNextRankPercentage() {Rai
```
Why?

AlexcastroDev commented:
@byronbytes I think this PR is have many changes and some comments of waiting for oppinions.

What you think close this one and open small PRs ? so we can review faster. 

bests :)

byronbytes commented:
> @byronbytes I think this PR is have many changes and some comments of waiting for oppinions.
> 
> What you think close this one and open small PRs ? so we can review faster.
> 
> bests :)

Sounds like a good idea! I've been pretty busy lately but I'll start with the smaller PR's eventually.
